tests/test_csc.py

```diff
@@ -0,0 +1,136 @@
+                self.assertEqual(12, event.limitXYMax)
+                self.assertEqual(-6, event.limitZMin)
+                self.assertEqual(12, event.limitZMax)
+                self.assertEqual(8, event.limitUVMax)
+                self.assertEqual(-6, event.limitWMin)
+                self.assertEqual(6, event.limitWMax)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_move_to_position(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_moveToPosition.set_start(timeout=STD_TIMEOUT, x=4, y=6, z=3, u=4,
+                                                                  v=3, w=6)
+                await asyncio.sleep(SHORT_TIMEOUT)
+                event = await harness.remote.evt_inPosition.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(False, event.inPosition)
+                event = await harness.remote.evt_inPosition.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(True, event.inPosition)
+                event = await harness.remote.evt_positionUpdate.aget(timeout=STD_TIMEOUT)
+                self.assertEqual(4, event.positionX)
+                self.assertEqual(6, event.positionY)
+                self.assertEqual(3, event.positionZ)
+                self.assertEqual(4, event.positionU)
+                self.assertEqual(3, event.positionV)
+                self.assertEqual(6, event.positionW)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_set_max_system_speeds(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_setMaxSystemSpeeds.set_start(timeout=STD_TIMEOUT, speed=1)
+                event = await harness.remote.evt_settingsAppliedVelocities.next(flush=False,
+                                                                                timeout=STD_TIMEOUT)
+                self.assertEqual(1, event.systemSpeed)
+
+        asyncio.get_event_loop().run_until_complete(doit())
+
+    def test_apply_position_offset(self):
+        async def doit():
+            async with Harness(initial_state=salobj.State.STANDBY) as harness:
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+                await harness.remote.cmd_applyPositionOffset.set_start(timeout=STD_TIMEOUT, x=1, y=1, z=1,
+                                                                       u=1, v=1, w=1)

```

Comment:
There is no event that actually verifies the Offset.

Reply:
I assume there is an event or telemetry topic that outputs the length of each actuator. You can check that.

If that does not exist then I strongly suggest you add one because we need to be able to monitor these kinds of commands. (Some of our early XML relied on users reading commands to keep track of some state. An example is the CCW monitoring rotator commands. That has proven to be a disaster and we are moving away from it as fast as we can.)

In any case you can introspect the CSC to find out what it is doing if no event is available.

Reply:
Added positionUpdate push and check that the proper offsets are applied by grabbing the previous position and applying offset and then verifying the difference between the 2 matches the offset.

Reply:
Sounds great. Thanks!